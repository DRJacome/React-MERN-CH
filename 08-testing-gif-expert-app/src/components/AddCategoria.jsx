import React, { useState } from 'react'
import PropTypes from "prop-types";

/* Como parámetro se envía desestructurada la propiedad "{setCategorias}" para 
 * establecer comunicación entre el padre "GifExpertApp" y el hijo "AddCategoria"
*/
export const AddCategoria = ({ setCategorias }) => {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    e.preventDefault()

    if (inputValue.trim().length > 2) {
      /* setCategorias(cat => [inputValue, ...cat]) */
      setCategorias(cat => [inputValue, ...cat])
      setInputValue('')
    }

    //console.log('Submit hecho');
  }
  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type='text'
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

/* PropTypes se usa para establecer reglas de requerimientos para la aplicación.
 * En este caso se usa "setCategorias: PropTypes.func.isRequired" para marcar una
 * regla que requiera que la función "setCaterorias" debe ser enviada desde 
 * el padre "GifExpertApp".
 */
AddCategoria.propTypes = {
  setCategorias: PropTypes.func.isRequired
  /* setCategorias: PropTypes.func */
}