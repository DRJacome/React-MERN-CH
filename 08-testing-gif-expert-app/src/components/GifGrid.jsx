import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from "./GifGridItem"
//import { getGifs } from "../helpers/getGifs"

export const GifGrid = ({ categoria }) => {
  const { data: imagenes, loading } = useFetchGifs(categoria)

  /*   useEffect(() => {
      getGifs(categoria)
        .then(
        )
      /*  El segundo argumento del useEffect es llamado "Lista/array de dependencias []" y 
      * y objetivo es controlar cuándo se renderiza el componente. Si el array está vacío,
      * el componente sólo se renderizará por primera vez, evitando que se renderice continuamente.
      
    }, [categoria])
   */

  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{categoria}</h3>
      {loading && <p className='carta animate__animated animate__flash'>Cargando</p>}
      {<div className="contenedorCartas">
        {imagenes.map(img => (
          <GifGridItem
            key={img.id}
            {...img}
          />
        ))}
      </div>}
    </>
  )
}

GifGrid.propTypes = {
  categoria: PropTypes.string.isRequired
}