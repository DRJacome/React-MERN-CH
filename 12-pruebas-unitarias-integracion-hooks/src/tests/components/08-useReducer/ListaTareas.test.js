import React from 'react'
import { shallow } from 'enzyme'
import { ListaTareas } from '../../../components/08-use-reducer/components/ListaTareas';
import { demoTareas } from '../../fixtures/demoTareas';

describe('Pruebas en <TodoList/>', () => {

  const gestionarBorrado = jest.fn()
  const gestionarTachado = jest.fn()

  const wrapper = shallow(
    <ListaTareas
      tareas={demoTareas}
      gestionarBorrado={gestionarBorrado}
      gestionarTachado={gestionarTachado}
    />);

  test('Debe mostrar el componente correctamente', () => {

    expect(wrapper).toMatchSnapshot();
  })

  test('Debe tener dos componente <ListaTareasItem/>', () => {

    console.log(wrapper.find('ListaTareasItem').length);
    console.log(demoTareas.length);

    expect(wrapper.find('ListaTareasItem').length).toBe(demoTareas.length);

    console.log(wrapper.find('ListaTareasItem').at(0).prop('gestionarBorrado'))
    expect(wrapper.find('ListaTareasItem').at(0).prop('gestionarBorrado')).toEqual(expect.any(Function));
  })

})