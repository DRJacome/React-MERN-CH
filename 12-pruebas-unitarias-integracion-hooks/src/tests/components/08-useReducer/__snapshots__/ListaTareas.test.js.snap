// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList/> Debe mostrar el componente correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <ListaTareasItem
    gestionarBorrado={[MockFunction]}
    gestionarTachado={[MockFunction]}
    index={0}
    key="1"
    tarea={
      Object {
        "descripcion": "Aprender React",
        "hecho": false,
        "id": 1,
      }
    }
  />
  <ListaTareasItem
    gestionarBorrado={[MockFunction]}
    gestionarTachado={[MockFunction]}
    index={1}
    key="2"
    tarea={
      Object {
        "descripcion": "Aprender Mongo",
        "hecho": false,
        "id": 2,
      }
    }
  />
</ul>
`;
